tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+
+    $collection = $this->createMock(OAuth2ManagerInterface::class);
+    $this->assertNull($collection->getClient());
+    $this->assertTrue(
+      method_exists($collection, 'setClient'),
+        'OAuth2ManagerInterface does not have setClient function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getClient'),
+        'OAuth2ManagerInterface does not have getClient function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getAccessToken'),
+        'OAuth2ManagerInterface does not have getAccessToken function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'setAccessToken'),
+        'OAuth2ManagerInterface does not have setAccessToken function/method'
+      );
+    $this->assertTrue(
+    method_exists($collection, 'getAuthorizationUrl'),
+        'OAuth2ManagerInterface does not have getAuthorizationUrl function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getState'),
+        'OAuth2ManagerInterface does not have getState function/method'
+      );
+    $this->assertTrue(
+      method_exists($collection, 'getUserInfo'),
+        'OAuth2ManagerInterface does not have getUserInfo function/method'
+      );
+
+  }
+
+  /**
+   * test for class SocialApiDataHandler
+   */
+
+  public function testSocialApiDataHandler () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/SocialApiDataHandler.php');
+    $this->session = $this->getMock(SessionInterface::class);
+    $collection = $this->getMockBuilder('Drupal\social_api\SocialApiDataHandler')
+                       ->setConstructorArgs(array($this->session))
+                       ->getMockForAbstractClass();
+
+    // checking for correct setSessionPrefix and getSessionPrefix method
+    $collection->setSessionPrefix('1234');
+    // var_dump($collection->getSessionPrefix());
+    $this->assertEquals('1234_', $collection->getSessionPrefix());
+  }
+
+  /**
+   * test for class SocialApiController
+   */
+
+  public function testSocialApiController () {
+
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/Controller/SocialApiController.php');
+
+    $this->namespaces = $this->createMock(Traversable::class);
+    $this->cache_backend = $this->createMock(CacheBackendInterface::class);
+    $this->module_handler = $this->createMock(ModuleHandlerInterface::class);
+
+    // passing parameters for constructor method in NetWorkManager Class
+   $this->networkManager = $this->getMockBuilder('Drupal\social_api\Plugin\NetworkManager')
+                                 ->setConstructorArgs(array($this->namespaces, $this->cache_backend, $this->module_handler))
+                                 ->getMock();
+
+
+   $collection = $this->getMockBuilder('Drupal\social_api\Controller\SocialApiController')
```

Comment:
Is using the name `$collection` a common thing? If so, can you point to examples in core? If not, please give them descriptive names (e.g. `$controller`)

Reply:
We can use any name.